### PR TITLE
Python(3) fathead

### PR DESCRIPTION
**Pull request by gaulrobe**
This is a first iteration that covers ONLY python 3 functions that were parsed out of the complete python 3.4 document set.

I think this would subsume the existing IA: https://duck.co/ia/view/python in that it has more coverage. I have not yet made an IA page for this, so I would like to know what other people thought. **New IA or subsume the existing one?**

I've left the output.txt file in here for now for people to give it a quick glance.

It'd be nice to get this out the door before any additional features are added to it so that we get a bump in coverage, and other people could iterate over it if ideas are had.

Future iterations could:
- Parse out class data
- Link to python 2 docs
- Could add a python2 document parser
- Wire this up to the python code searching IA that @jrasanen is working on

@jdorweiler @jrasanen @moollaza @preemeijer @pjhampton

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jdorweiler** [comment]
Looks great @gaulrobe. Just one comment.
Could you separate the class name from the method or create redirects for these entries? For example:

"addpackagepath" doesn't work
https://beta.duckduckgo.com/?q=addpackagepath&ia=about

"modulefinder.addpackagepath" doesn't work
https://beta.duckduckgo.com/?q=modulefinder.addpackagepath&ia=about

"modulefinder addpackagepath" works
https://beta.duckduckgo.com/?q=modulefinder+addpackagepath&ia=about

**Comment by gaulrobe** [comment]
@jdorweiler Thanks, I will look into this!

File: lib/fathead/python/parse.py [hunk 1]
```diff
@@ -221,13 +226,19 @@ def create_file(self):
...
+                    if redirect == name:
+                        redirect = ''
+
                     abstract = '{}{}{}'.format(method_signature, '<br>' if method_signature and first_paragraph else '', first_paragraph)
                     url = data_element.get('url')
                     list_of_data = [
                         name,                       # unique name
                         'A',                        # type is article
-                        '',                         # no redirect data
+                        redirect,                   # no redirect data
```
Comment (jdorweiler) [inline lib/fathead/python/parse.py]: You'll have to change the type to "R" if this is a redirect and leave the other fields blank. So a redirect entry will look like this:
```
module.function    R    function
````

Comment (gaulrobe) [inline lib/fathead/python/parse.py]: @jdorweiler 

Ah, to clarify then, there would be:
- 1 entry for the original `module.function`
- N entries for any redirect aliases needed for a specific `module.function`

Is that accurate?

Comment (jdorweiler) [inline lib/fathead/python/parse.py]: Exactly

**Comment by gaulrobe** [comment]
@jdorweiler OK, I think I have the redirect aliases on here correctly. Another look?

File: lib/fathead/python/parse.py [hunk 2]
```diff
@@ -252,6 +247,25 @@ def create_file(self):
                     ]
                     output_file.write('{}\n'.format('\t'.join(list_of_data)))
 
+                    # Add redirect if we got a redirect name that is different from the original name
+                    if redirect != name:
+                        list_of_data = [
+                            name,                       # unique name
+                            'R',                        # type is redirect
+                            redirect,                   # redirect alias
```
Comment (jdorweiler) [inline lib/fathead/python/parse.py]: Other way around.  `name    R    redirect`

**Comment by jdorweiler** [comment]
Very close. Just one small fix.

File: lib/fathead/python/parse.py (hunk 2 above)
Comment (gaulrobe) [inline lib/fathead/python/parse.py]: D'oh! That makes sense... name has to be unique! 😰

**Comment by gaulrobe** [comment]
@jdorweiler OK... I _think_ I got this now 😄

**Comment by jdorweiler** [comment]
Success!  https://beta.duckduckgo.com/?q=modulefinder+AddPackagePath&ia=about

There were still a few entries that had new lines in their code blocks. I was just fixing it manually in vim `s/<code>\n/<code>/`. 

Want to fix that and give it one last check on beta?

**Pull request merged**